vspec2x_abstract.py

```diff
@@ -0,0 +1,172 @@
+
+
+# Copyright (c) 2016 Contributors to COVESA
+#
+# This program and the accompanying materials are made available under the
+# terms of the Mozilla Public License 2.0 which is available at
+# https://www.mozilla.org/en-US/MPL/2.0/
+#
+# SPDX-License-Identifier: MPL-2.0
+
+#
+# Convert vspec files to various other formats
+#
+
+from vspec.model.vsstree import VSSNode
+from vspec.model.constants import VSSTreeType
+from vspec.loggingconfig import initLogging
+import argparse
+import logging
+import sys
+import vspec
+
+parser = argparse.ArgumentParser(description="Convert vspec to other formats.")
+
+
+class Vspec2XAbstract:
+
+    def __init__(self, exporter):
+
+        self.exporter = exporter
+
+    def main(self, arguments):
+        initLogging()
+
+        parser.add_argument('-I', '--include-dir', action='append', metavar='dir', type=str, default=[],
```

Comment:
I think it would be good to change logic, that somehthing like vspec2json "owns" the parser for command line arguments and can include/instanctiate vpsec2x with a feature array (do I know instances/structs/uuid/extended attributes), and then vspec2x would "add" generic options , and leave options out that a tool does not support anyway

Reply:
It would be quite easy to do some changes of the current solution. Today we have if , if looking for instance at [vss2json.py](https://github.com/COVESA/vss-tools/blob/master/vspec/vssexporters/vss2json.py#L21) two methods:

```
def feature_supported(feature_name: str)
def add_arguments(parser: argparse.ArgumentParser):
````

Where `add_arguments` is for adding exporter-specific arguments,  and ` feature_supported` to answer yes/no on features like "no expand" (for struct we have a different mechanism), but I agree that passing in some form of config array would be a nicer solution, as we then could exclude non-supported arguments already when you do `-h`.